### PR TITLE
ACCUMULO-4153: Update the getCodec method to no longer be synchronize…

### PR DESCRIPTION
**Pull request by phrocker**
…d using static initializer in enum

I forgot about this PR. I made changes and ran them in production. they work well. it's simply allocating these objects with a static initializer in the enum. I hastily typed this from my other machine, so please tell me if I did something stupid.

### CODE REVIEW

File: core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+  }
+
+  // don't start until we have created the codec
+  @Test
+  public void testThereCanBeOnlyOne() throws IOException, InterruptedException {
+
+    ExecutorService service = Executors.newFixedThreadPool(20);
+
+    Collection<Callable<Boolean>> list = Lists.newArrayList();
```
Comment (ctubbsii) [inline core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java]: This line, and the newHashSet one below it breaks the build (modernizer plugin). Prefer to use Java 7+ diamond operator.

Comment (phrocker) [inline core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java]: My apologies I must not have run the right profile . I'll make sure I fix and run modernizer with he appropriate profile

Comment (phrocker) [inline core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java]: Does it not execute on clean install? I'll check when I get home I have a mod maven build and profile . May have pooged myself

Comment (keith-turner) [inline core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java]: try `mvn clean verify -DskipITs`

**Comment by keith-turner** [comment]
I just took a quick look at this, I am going to look at it more in the morning.  One thing I dont understand is that all the impl createCompressionStream and createDecompressionStream methods are synchronized. Maybe that because they are mutating the conf, but it seems the conf is scoped wider than the sync.  Also, is this a bottleneck?  I will look into it more in the morning.

**Comment by phrocker** [comment]
I left them synchronized because conf is not thread safe. I left that paradigm the same because it was not a bottleneck in profiling. Creating a conf per use would have been.

File: core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java (hunk 1 above)
Comment (phrocker) [inline core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java]: thanks!

Comment (phrocker) [inline core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java]: clean verify didn't run it. I'll take a look tomorrow, perhaps. and I have nothing custom profile in this vm...

Comment (phrocker) [inline core/src/test/java/org/apache/accumulo/core/file/rfile/bcfile/CompressionTest.java]: so it turns out I got it to run with compile. If you notice my branch targets 1.6.6, but I did the PR to master unintentionally. I tried verify in master and it runs modernizer. sorry for the confusion. I'm an idiot.

**Pull request closed without merging**